core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -266,6 +273,9 @@ class Partition(val topicPartition: TopicPartition,
   newGauge("AtMinIsr", () => if (isAtMinIsr) 1 else 0, tags)
   newGauge("ReplicasCount", () => if (isLeader) assignmentState.replicationFactor else 0, tags)
   newGauge("LastStableOffsetLag", () => log.map(_.lastStableOffsetLag).getOrElse(0), tags)
+  ISR_STATES_TO_CREATE_METRICS.foreach(c => newGauge(s"${c.getSimpleName}", () => if (isrStateClass.equals(c)) 1 else 0, tags))
+
+  def isrStateClass: Class[_ <: IsrState] = isrState.getClass
```

Comment:
Why not put it inline in l276? e.g. "isrState.getClass.equals(c)"

Reply:
This is... more of a personal taste.  The way you mentioned is not wrong either.

I feel like "making comparison on the state", or a method `.isIsrStateOfClass(Class[_ <: IsrState])` is an over extended functionality to this class, but just exposing the inspection is still within.  So I decided to make it encapsulated in the Gauge logic.

Reply:
Oh now I see your point.  In terms of monitoring, we should monitor on the states of `Pending`s.  If it is raise too high (e.g. 30% of leader replicas), then it is problematic and requires manual intervention (e.g. restart)